packages/sanity/src/core/preview/components/PreviewLoader.tsx

```diff
@@ -27,21 +27,31 @@ export function PreviewLoader(
     component: ComponentType<Omit<PreviewProps, 'renderDefault'>>
   }
 ): ReactElement {
-  const {layout, value, component, style: styleProp, schemaType, ...restProps} = props
+  const {
+    layout,
+    value,
+    component,
+    style: styleProp,
+    schemaType,
+    __internal_skip_visibility_check,
+    ...restProps
+  } = props
 
   const [element, setElement] = useState<HTMLDivElement | null>(null)
-  const isPTE = layout && ['inline', 'block', 'blockImage'].includes(layout)
+  const shouldSkipVisbility =
```

Comment:
While discussing with Bjørge we spoke about removing this from the component and replacing this in whichever parts made sense in PTE (given the name of the const)

However, after spending some more time on it, I realised that it's not only PTE situations where this would be an issue. The naming itself i misleading because while it probably has most import in PTE it also means that, if we were to remove this, if I was to create a preview element with these layouts they would start flickering as well which I don't think is what we want to do.

Given this I renamed the const

Reply:
After discussing this a bit with @bjoerge we made the decision that the fact that the layout was impacting the "visibility" is incidental and it shouldn't impact it in the way that it does. After some investigation we found the places to make sure that PTE would work as before despite the changes to the PreviewLoader (cc @skogsmaskin )

860a6e39fd80375cdacf934039597b4d77618e5c

It is important that if people are creating and using the <preview> in the way that was given as an example that they might need to refactor the component or simply use the new provided prop. The layouts impacting the way they did shouldn't be taken into account.